examples/room/lod/lodmanager/manager.go

```diff
@@ -0,0 +1,289 @@
+	ver                   uint8
+	cmd                   uint16
+	syncIntervalForNearby int32
+	syncIntervalForFar    int32
+	maxDistanceForNearby  int32
+	maxDistanceForFar     int32
+	managerMapMutex       sync.RWMutex
+	sendBuffer            chan sendMessage
+	senderWg              sync.WaitGroup
+}
+
+func (m *Manager) String() string {
+	m.managerMapMutex.RLock()
+	defer m.managerMapMutex.RUnlock()
+	return fmt.Sprintf("Manager{userEntities: %v, started: %v, ver: %v, cmd: %v, syncIntervalForNearby: %v, syncIntervalForFar: %v,	 maxDistanceForNearby: %v, maxDistanceForFar: %v}",
+		m.userEntities, m.started.Load(), m.ver, m.cmd, m.syncIntervalForNearby, m.syncIntervalForFar, m.maxDistanceForNearby, m.maxDistanceForFar)
+}
+
+// NewManager creates a new LOD manager with the specified configuration
+func NewManager(ver uint8, cmd uint16, syncIntervalForNearby int32, syncIntervalForFar int32, maxDistanceForNearby int32, maxDistanceForFar int32) *Manager {
+	// Initialize metrics on first manager creation
+
+	lm := &Manager{
+		ver:                   ver,
+		cmd:                   cmd,
+		syncIntervalForNearby: syncIntervalForNearby,
+		syncIntervalForFar:    syncIntervalForFar,
+		maxDistanceForNearby:  maxDistanceForNearby,
+		maxDistanceForFar:     maxDistanceForFar,
+		userEntities:          make(map[string]*UserEntity),
+		started:               atomic.Bool{},
+		sendBuffer:            make(chan sendMessage, 10000), // Buffer size: 10000
+	}
+
+	lm.started.Store(true)
+
+	// Increment instance count
+	instancesGauge.Inc()
+
+	// Start send worker goroutine
+	lm.senderWg.Add(1)
+	go lm.sendWorker()
+
+	// Start LOD loop
+	go lm.invokeLodLoop()
+
+	return lm
+}
+
+// AddUserEntity adds or updates a user entity with position and payload data
+func (m *Manager) AddUserEntity(userID string, x int32, y int32, payload []byte) {
+	m.managerMapMutex.Lock()
+	defer m.managerMapMutex.Unlock()
+	if userEntity, ok := m.userEntities[userID]; ok { // update
+		userEntity.Payload = payload
+		userEntity.X = x
+		userEntity.Y = y
+		userEntity.ChangedAfterSend = true
+		return
+	}
+	m.userEntities[userID] = NewUserEntity(x, y, payload)
+	// Update active users gauge
+	activeUsersGauge.Set(int64(len(m.userEntities)))
+}
+
+// RemoveUserEntity removes a user entity from the manager
+// and remove remember data of other user entities
+func (m *Manager) RemoveUserEntity(userID string) {
+	m.managerMapMutex.Lock()
+	defer m.managerMapMutex.Unlock()
+	if _, ok := m.userEntities[userID]; !ok {
+		return
+	}
+	delete(m.userEntities, userID)
+	// delete from others remember data
+	for _, userEntity := range m.userEntities {
+		userEntity.RememberMutex.Lock()
+		delete(userEntity.Remember, userID)
+		userEntity.RememberMutex.Unlock()
+	}
+	// Update active users gauge
+	activeUsersGauge.Set(int64(len(m.userEntities)))
+}
+
+// sendWorker processes messages from sendBuffer and sends them to users
+func (m *Manager) sendWorker() {
+	defer m.senderWg.Done()
```

Comment:
IMO: If you use a channel (done) here, you can eliminate the uneven wait group usage.

Reply:
That would look something like this:
```suggestion
	for {
		select {
		case <-done:
			return
		case msg := <-m.sendBuffer:
			receiverUser := user.GetUserBySID(msg.receiverUserID)
			if receiverUser != nil {
				receiverUser.PushToClient(msg.ver, msg.cmd, msg.payload, packet.Unreliable)
				// Track successfully sent messages
				messagesSentCounter.Inc()
			} else {
				// Track errors
				errorsCounter.WithLabelValues("user_not_found").Inc()
			}
		}
	}
```

Reply:
changed to use channel from working Group!